common/src/app/common/files/variant.cljc

```diff
@@ -61,3 +62,22 @@
     (and (seq shapes)
          (not= (:main-instance-id component) (last shapes)))))
 
+(defn get-primary-variant
+  [data component]
+  (let [page-id    (:main-instance-page component)
+        objects    (-> (dm/get-in data [:pages-index page-id])
+                       (get :objects))
+        variant-id (:variant-id component)]
+    (->> (dm/get-in objects [variant-id :shapes])
+         last
```

Comment:
Shapes is known to be a vector, in this case the most efficient way to get the last element is using `peek`